botcore/async_stats.py

```diff
@@ -0,0 +1,59 @@
+"""An async transport method for statsd communication."""
+
+import asyncio
+import socket
+from typing import Optional
+
+from discord.utils import MISSING
+
+from botcore.utils import scheduling
+
+try:
+    from statsd.client.base import StatsClientBase
+except ImportError:
+    StatsClientBase = MISSING
```

Comment:
This could be more robust, but I am not sure if it is worth it 
```suggestion
try:
    import statsd
except ImportError:
    StatsClientBase = MISSING
else:
    from statsd.client.base import StatsClientBase
```

Reply:
https://github.com/python-discord/bot-core/pull/42/commits/49df21e48da18c01c6bea1cbb9ce1187e89adcf3

Reply:
Making statsd optional has been dropped now, see comment above.